packages/notifi-graphql/lib/NotifiSubscriptionService.ts

```diff
@@ -0,0 +1,82 @@
+import { createClient, Client } from 'graphql-ws';
+export class NotifiSubscriptionService {
+  public wsClient: Client | undefined;
+  constructor(private wsurl: string) {
+  }
+
+  disposeClient = () => {
+    if (this.wsClient) {
+      this.wsClient.terminate();
+      this.wsClient.dispose();
+    }
+  };
+
+  subscribe = (jwt: string | undefined, subscriptionQuery: string, actionOnMessageReceived: () => void | undefined) => {
+    this.initializeClientIfUndefined(jwt);
+    this.wsClient?.subscribe({
+      query: subscriptionQuery,
+      extensions: {
+        type: 'start'
+      }
+    },
+      {
+        next: (data) => {
+          if (actionOnMessageReceived) {
+            actionOnMessageReceived();
+          }
+          console.log("Subscription data recieved" + JSON.stringify(data));
```

Comment:
Instead of printing out the `data`, shall we put the data into the callback so that clients have more control? ex below
```typescript
onNext?: (data) => void
```

Reply:
done